### PR TITLE
Generate nuget packages for Arm64 build

### PR DESCRIPTION
**Pull request by gkhanna79**
@weshaggard @ericstj @pgavlin @rahku  PTAL

CC @haripulapaka

### CODE REVIEW

**Comment by gkhanna79** [comment]
Fixes https://github.com/dotnet/coreclr/issues/3080

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT arm64 Release

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT arm64 Checked please

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT arm64 Release please

**Comment by rahku** [comment]
Thanks. native arm64 crossgen seems missing from the package. If it is not placed in the package we will lose it or it needs to be re-built everytime we need it.

**Comment by gkhanna79** [comment]
> native arm64 crossgen seems missing from the package.

For now, I have intentionally kept it out since it is not expected to be used in production scenarios (i.e. we expect to generate native images in cross-compilation scenario and the crossgen under tools\x64_arm64 should suffice).

Is there a specific scenario you have in mind?

**Comment by rahku** [comment]
Yes in order to be able to run R2R tests on the device we need native crossgen because crossgen is run as part of runtests.cmd

**Comment by rahku** [comment]
Should be also required to run R2R helix runs on the target.

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT arm64 Checked please

**Comment by gkhanna79** [comment]
@rahku Sounds good. I have added native crossgen to the package as well.

File: src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <Version>1.0.3</Version>
     <SkipPackageFileCheck>true</SkipPackageFileCheck>
     <SkipValidatePackage>true</SkipValidatePackage>
-    <PackagePlatforms>x64;x86;arm</PackagePlatforms>
+    <PackagePlatforms>x64;x86;arm64;arm</PackagePlatforms>
```
Comment (weshaggard) [inline src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]: orthogonal but something I notices while reviewing. This claims to support arm but I don't see an arm configuration below.

File: dir.props [hunk 2]
```diff
@@ -144,7 +144,7 @@
     <PackageThirdPartyNoticesFile>$(SourceDir).nuget/ThirdPartyNotices.txt</PackageThirdPartyNoticesFile>
 
     <!-- This should be kept in sync with package details in src/.nuget/init/project.json -->
-    <RuntimeIdGraphDefinitionFile>$(PackagesDir)/Microsoft.NETCore.Platforms/1.0.1-rc2-23712/runtime.json</RuntimeIdGraphDefinitionFile>
+    <RuntimeIdGraphDefinitionFile>$(PackagesDir)/Microsoft.NETCore.Platforms/1.0.2-beta-24224-02/runtime.json</RuntimeIdGraphDefinitionFile>
```
Comment (weshaggard) [inline dir.props]: It is probably worth adding something that will validate cause this to get updated.

**Comment by weshaggard** [comment]
LGTM

**Comment by rahku** [comment]
LGTM

File: src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj (hunk 1 above)
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]: I dont think it claims to support arm per https://github.com/dotnet/buildtools/blob/3c700d1ecc358c6c1b3ffbaaeb49e7864a8e9ba7/src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets#L125 which expects a trailing ";" to enable the platform.

That said, I will be looking into enabling Arm32 once I am done with this.

File: dir.props (hunk 2 above)
Comment (gkhanna79) [inline dir.props]: Like a bot update?

Comment (weshaggard) [inline dir.props]: Yes a bot update but also a validation task similar to what we do in corefx to ensure this version agrees with what is in your project.json.

Comment (gkhanna79) [inline dir.props]: Can you please point to it?

Comment (weshaggard) [inline dir.props]: What we do is define the expected version in an msbuild property like such https://github.com/dotnet/corefx/blob/master/dir.props#L101 and then we run the validation task https://github.com/dotnet/corefx/blob/master/build.proj#L108 that will verify the versions we have defined are the versions referenced in all our project.json files.

**Pull request merged**